inc/sitemaps/class-sitemap-image-parser.php

```diff
@@ -62,19 +48,22 @@ public function get_images( $post ) {
 
 		$images = array();
 
-		$content = $post->post_content;
-		$content = '<p><img src="' . $this->image_url( get_post_thumbnail_id( $post->ID ) ) . '" alt="' . $post->post_title . '" /></p>' . $content;
-
-		if ( preg_match_all( '`<img [^>]+>`', $content, $matches ) ) {
-
-			$images = $this->parse_matched_images( $matches[0], $post );
+		if ( ! is_object( $post ) ) {
```

Comment:
From this method, it is passed (line 65) into `$this->parse_html_images()` which calls `get_post_field()` with it, and that requires (via documentation - no type hinting here) `$post` to be an `int` or `WP_Post`.

Line 65 also passes `$post` to `$this->parse_galleries()`, which documents it as being `WP_Post` and calls `$this->get_post_galleries()` (which also documents it as being a `WP_Post`) which in turn requires and assumes it to have a `post_content` property.

I've not found where and how this `get_images()` method might be used (which is likely your concern here), but if the likelihood is it's only used internally, I don't see why the checks aren't a little stricter earlier on in the logic. There's little performance difference between creating a `stdClass` and `WP_Post`.